pkg/tsdb/influxdb/response_parser.go

```diff
@@ -36,15 +36,17 @@ func (rp *ResponseParser) Parse(buf io.ReadCloser, query *Query) *backend.QueryD
 		return resp
 	}
 
-	frames := data.Frames{}
-	for _, result := range response.Results {
-		frames = append(frames, transformRows(result.Series, query)...)
+	for i, result := range response.Results {
```

Comment:
there are changes here that are not strictly backward compatibly.
they might be totally ok, i just want to mention it.

you see, in the old code, as long as  `response.Error == ""`, we transformed all the `response.Results` into dataframes, and returned those dataframes at the end.
and, if in one of `response.Results` there was an `Error != ""`, we stored that in `queryRes.Error` (yes, possibly overwriting previous values there), and returned that too.

in the new version, if we reach a `response.Results` array item, that has `Error`, we immediately return with all the dataframes created up to that point, but we do not extract dataframes from the current array item, and also we don't extract them from the remaining array items.

i'd look at these questions:
1. how is this handled in the typescript version?
2. what is more correct?

Reply:
Thanks for the tip. Now we don't return if there is an error. We add the error to our response and continue. I think if there is an error the user should be alerted (via the UI error) and we should not try to process the query that has the error. 

Even though we only send the last error right now (see in pic) as specified by `DataQueryError` , the user will continue to get an error on the UI (one at a time) until all errors have been cleared. This seems to be the way it works for all datasources.

![Screenshot 2022-01-24 at 09 35 47](https://user-images.githubusercontent.com/90795735/150758331-5ac1408f-6304-44fd-9971-2cdbca60fe6c.png)

Reply:
nice. in general, when working with backend datasources, i think (pinging @aocenas to doublecheck this) we want to do this logic:
- when we get back a response from the remote database, the response is correctly formatted, and inside the response it says "your query-text was wrong", we set the error attribute inside the correct  `DataResponse` object
- if the error happened somewhere else, like, we were unable to open a http-connection to the remote database, or something else, we return an `error` from `func QueryData`

Reply:
I see you resolved this convo, @gabor. I actually pressed the unresolve convo by mistake but I hear you and this is the way the errors are structured. Let me know if there is an action here.